Lib/idlelib/autocomplete.py

```diff
@@ -47,6 +38,10 @@ def __init__(self, editwin=None):
             self._delayed_completion_id = None
             self._delayed_completion_index = None
 
+    def reset(self):
+        self.popupwait = idleConf.GetOption("main", "General",
+                                   "autocomplete_wait", type="int", default=0)
+
```

Comment:
Built-in options are effective when they are changed from the config menu. Extensions require restarting IDLE. In order for these to function like built-ins, their changed settings needed to be reloaded from within the editor. That is what the reset function is for. There is one in each of the old extensions that had options. It seems the activation got lost when I was resolving conflicts, but it looks like the place to call it is now activate_config_changes in the configdialog. Would you rather I put the functionality back in, or that some of the new options behave differently than the old ones?

Reply:
I opened bpo-22707 two years ago for this problem, and intended that we work on it after this one. I just posted my analysis and made you nosy.  For reasons explained there, I want to leave the class attributes as class attributes.  Whether the idleConf wrapper are added here or in 22707, they should be called 'reload'; 'reset' implies changing back to initial values.  They should be classmethods called after the class statement.  This will preserve the current behavior, which I will manually test.

 The line length for ReformatParagraph should be a class attribute, and the test adjusted.  CodeContext().visible should be left alone until fixed or deleted.

I want to deal with changing code to update immediately in 22707, where the new code will be tested.  I will add more on that issue.